### PR TITLE
Added GitLab Authentication

### LINKED ISSUE #258: [bgw-net] Add Server authentication for frontend
Add Gitlab authentication for server frontend.

### PR DESCRIPTION
**Pull request by heaterscar**
Closes #258

### CODE REVIEW

File: bgw-net/bgw-net-server/src/main/kotlin/tools/aqua/bgw/net/server/service/oauth/SpringConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+package tools.aqua.bgw.net.server.service.oauth
+
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.builders.WebSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.web.util.matcher.RequestMatcher
+
+private const val LOGOUT_SUCCESS_URL = "/"
```
Comment (dominikmaeckel) [inline bgw-net/bgw-net-server/src/main/kotlin/tools/aqua/bgw/net/server/service/oauth/SpringConfiguration.kt:27]: Declare constants in separate file

File: bgw-net/bgw-net-server/src/main/kotlin/tools/aqua/bgw/net/server/service/oauth/SpringConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * Configures spring security, doing the following:
+ * - Bypass security checks for static resources,
+ * - Restrict access to the application, allowing only logged in users,
+ * - Set up the login form
+ */
+@Configuration
+@EnableWebSecurity
+class SecurityConfiguration() : WebSecurityConfigurerAdapter() {
+  /** Require login to access internal pages and configure login form. */
```
Comment (dominikmaeckel) [inline bgw-net/bgw-net-server/src/main/kotlin/tools/aqua/bgw/net/server/service/oauth/SpringConfiguration.kt:38]: Docs does not match what gets configured.

**Review by dominikmaeckel — CHANGES_REQUESTED** [review]
See comments.

File: bgw-net/bgw-net-server/src/main/kotlin/tools/aqua/bgw/net/server/service/oauth/SpringConfiguration.kt (hunk 1 above)
Comment (heaterscar) [reply bgw-net/bgw-net-server/src/main/kotlin/tools/aqua/bgw/net/server/service/oauth/SpringConfiguration.kt]: fixed

File: bgw-net/bgw-net-server/src/main/kotlin/tools/aqua/bgw/net/server/service/oauth/SpringConfiguration.kt (hunk 2 above)
Comment (heaterscar) [reply bgw-net/bgw-net-server/src/main/kotlin/tools/aqua/bgw/net/server/service/oauth/SpringConfiguration.kt]: fixed

**Review by dominikmaeckel — APPROVED** [review]

**Pull request merged**